teps/0095-repository-crd.md

```diff
@@ -0,0 +1,247 @@
+---
+status: proposed
+title: Repository CRD
+creation-date: '2021-11-19'
+last-updated: '2021-11-19'
+authors:
+- '@sbwsg'
+---
+
+# TEP-0095: Repository CRD
+---
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Several projects and designs currently in-flight in the Tekton ecosystem
+are related to or directly working with source repository-related information.
+These include:
+
+- [Scheduled and Polling Runs in Tekton](0083-scheduled-and-polling-runs-in-tekton.md)
+- [Workflows](../working-groups.md#workflows)
+- [Pipeline-as-Code](https://github.com/openshift-pipelines/pipelines-as-code)
+- [Remote Resource Resolution](0060-remote-resource-resolution.md)
+
+Each of these projects are dependent on information from
+repositories - either querying for metadata, fetching content, listening for
+events, recording relatational data about them or publishing notifications to
+them.
+
+Given a shared interest in this data across multiple projects it might
+make sense to define an object that allows an operator to specify common
+repo configuration only once, rather than per-project. Over time other projects
+and tools could also leverage this information.
+
+## Motivation
+
+### Goals
+
+- Decide if there is a common set of data that would be useful across
```

Comment:
Is this just for source code repositories as stated in the summary? or would you aim to handle e.g. image repositories as well?

Reply:
The original intention behind the TEP is solely around repository information. I've updated this goal to call that out specifically.